src/main/scala/rps/Main.scala

```diff
@@ -1,35 +1,69 @@
 package rps
-import scala.io.StdIn.readLine
 import scala.util.Random
 import io.buildo.enumero.annotations.enum
 import io.buildo.enumero.{CaseEnumIndex, CaseEnumSerialization}
 import Move._
+import Response._
 import Result._
-import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
 import io.circe.generic.auto._
 import io.buildo.enumero.circe._
 import akka.actor.ActorSystem
-import akka.http.scaladsl.Http
-import akka.http.scaladsl.model._
-import akka.http.scaladsl.server.Directives._
 import akka.stream.ActorMaterializer
-import scala.io.StdIn
+import wiro.annotation._
+import scala.concurrent.{ExecutionContext, Future}
+import wiro.Config
+import wiro.server.akkaHttp._
+import wiro.server.akkaHttp.FailSupport._
 
-object Main extends App {
-  WebServer.run
+@enum trait Move {
+  object Rock
+  object Paper
+  object Scissors
+}
+
+@enum trait Result {
+  object Win
+  object Lose
+  object Draw
+}
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import wiro.Config
+import wiro.server.akkaHttp._
+
+object WebServer extends App with RouterDerivationModule {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  implicit def throwableResponse: ToHttpResponse[Throwable] = null
+
+  val gameRouter = deriveRouter[GameApi](new GameApiImpl)
+
+  val rpcServer = new HttpRPCServer(
+    config = Config("localhost", 8080),
+    routers = List(gameRouter)
+  )
+}
+
+@path("rps")
+trait GameApi {
```

Comment:
in the next step, you will perform a 3-layer split of the application, but in the meantime you could create another file for GameApi and GameApiImpl